pkg/cli/server.go

```diff
@@ -396,5 +397,26 @@ func loadRemoteWriteTargetConfigsFromFile(c *config.Server) error {
 	}
 
 	c.RemoteWrite.Targets = s.RemoteWrite.Targets
+
+	// Setup defaults
+	// Since we are loading from yaml directly, viper's defaults won't work
+	// TODO(eh-am): find a better solution?
+	for k, t := range c.RemoteWrite.Targets {
+		if t.QueueSize == 0 {
+			t.QueueSize = 100
+		}
+
+		if t.QueueWorkers == 0 {
+			t.QueueWorkers = 4
+		}
```

Comment:
I think it should be at least `runtime.NumCPU`. Note that the task is IO bounded, it may make sense to set N workers per core.